h/api/test/auth_test.py

```diff
@@ -28,3 +31,150 @@ def test_translate_annotation_principals(p_in, p_out):
     result = auth.translate_annotation_principals(p_in)
 
     assert set(result) == set(p_out)
+
+
+generate_bearer_token_fixtures = pytest.mark.usefixtures('jwt')
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_calls_encode(jwt):
+    """It should pass the right arguments to encode()."""
+    before = datetime.datetime.utcnow()
+    request = mock_request()
+
+    auth.generate_bearer_token(request, 3600)
+
+    assert jwt.encode.call_args[0][0]['sub'] == 'acct:testuser@hypothes.is', (
+        "It should encode the userid as 'sub'")
+    after = datetime.datetime.utcnow() + datetime.timedelta(seconds=3600)
```

Comment:
Hmm. I like the reusability of `approximate_matcher()`. But if `generate_bearer_token()` took longer than two seconds to execute that would fail. Not that I think it's likely to take that long, but I like the fact that `before` is definitely a time before it was called, and `after` + 3600 seconds should definitely be a time after the expiry time, it's not dependent on any timing of how long things take to run.

What's hard to maintain about it?